akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorStashSpec.scala

```diff
@@ -571,14 +572,15 @@ class EventSourcedBehaviorStashSpec
     }
 
     "discard when stash has reached limit with default dropped setting" in new StashLimit {
-
+      override val customStashLimit: Option[Int] = None
       val c = spawn(behavior)
 
       // make sure it completed recovery, before we try to overfill the stash
       c ! "ping"
       probe.expectMessage("pong")
 
       c ! "start-stashing"
+      probe.expectMessage("started-stashing")
```

Comment:
These tests triggered started-stashing and then piled up messages without waiting for that to complete, so they end up in the non-explicit stash because of the write being in progress leading to the "unstash" message also having to be put in the stash buffer, which is at that point full.

Reply:
I would add this comment to the source code :)

Reply:
Added in [ee983a6](https://github.com/akka/akka/pull/31808/commits/ee983a68d0372ada1e59760803f80e2670394587)